### PR TITLE
Store only best feasible solutions

### PR DESCRIPTION
**Pull request by leonlan**
This PR fixes a bug where infeasible best solutions were collected.

### CODE REVIEW

File: pyvrp/GeneticAlgorithm.py [hunk 1]
```diff
@@ -155,7 +155,10 @@ def _educate(self, individual: Individual):
                     ):
                         self._ls.intensify(individual)
 
-                    if individual.cost() < self._best.cost():
+                    if (
+                        individual.is_feasible()
+                        and individual.cost() < self._best.cost()
+                    ):
                         self._best = individual
```
Comment (N-Wouda) [inline pyvrp/GeneticAlgorithm.py:162]: It's not needed here because this already happens inside `if individual.is_feasible()`

**Review by N-Wouda — COMMENTED** [review]
Thanks! Can you update the test in `test_GeneticAlgorithm.py` to explicitly check that the returned solutions are feasible?

File: pyvrp/GeneticAlgorithm.py (hunk 1 above)
Comment (leonlan) [reply pyvrp/GeneticAlgorithm.py:162]: This might still be needed because an intensified solution can become infeasible I think.

Comment (N-Wouda) [reply pyvrp/GeneticAlgorithm.py:162]: Yeah that's fair. But then the register calls below this are also wrong. Can you update those?

**Review by N-Wouda — APPROVED** [review]

**Pull request merged**